docs/customization/factory.rst

```diff
@@ -0,0 +1,131 @@
+
+Why would you customize a Factory?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Differently configured versions of resources may be needed in various scenarios in your application.
+You may need for instance to:
+
+    * create Product with Supplier(which is your own custom entity)
+    * create a disabled Product(for further modifications)
+    * create a ProductReview with predefined description
+
+and many, many more usecases.
+
+How to customize a Factory?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would want to have a possibility to create disabled products.
+
+1. Create your own factory class in the ``AppBundle\Factory`` namespace.
+Remember that it has to extend a proper base class. How can you check that?
+
+For the ``ProductFactory`` run:
+
+.. code-block:: bash
+
+    $ php bin/console sylius:debug:resource sylius.product
+
+As a result you will get a table od Product related classes. Check the ``classes.factory`` row
+where you will find the ``Sylius\Component\Product\Factory\ProductFactory`` - this is the class that you need to be decorating.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Factory;
+
+    use Sylius\Component\Core\Model\ProductInterface;
+    use Sylius\Component\Product\Factory\ProductFactoryInterface;
+
+    class ProductFactory implements ProductFactoryInterface
+    {
+        /**
+         * @var FactoryInterface
+         */
+        private $decoratedFactory;
+
+        /**
+         * @param ProductFactoryInterface $factory
+         */
+        public function __construct(ProductFactoryInterface $factory)
+        {
+            $this->decoratedFactory = $factory;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function createNew()
+        {
+            return $this->decoratedFactory->createNew();
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function createWithVariant()
+        {
+            return $this->decoratedFactory->createWithVariant();
+        }
+
+        /**
+         * @return ProductInterface
+         */
+        public function createDisabled()
+        {
+            /** @var ProductInterface $product */
+            $product = $this->decoratedFactory->createNew();
+
+            $product->setEnabled(false);
+
+            return $product;
+        }
+    }
+
+2. In order to decorate the base ProductFactory with your implementation you need to configure it
+as a decorating service in the ``AppBundle\Resources\config\services.yml``.
+
+.. code-block:: yaml
+
+    services:
+        app.factory.product:
+            class: AppBundle\Factory\ProductFactory
+            decorates: sylius.factory.product
+            arguments: ['@app.factory.product.inner']
+            public: false
+
+3. After the ``sylius.factory.product`` has been decorated it has been extended by the new ``createDisabled()`` method.
```

Comment:
Points **1.** and **2.** tell people what needs to be done, this one however is purely informational. Would be  more natural, if lines 104 and 106 were switched, with a slight rephrasing.